### PR TITLE
Use RMN changeset in e2e tests and refactor RMN remote changeset

### PR DESCRIPTION
**Pull request by carte7000**
Use RMN changeset in e2e tests and refactor RMN remote changeset

RMN remote changeset was changed to allow more flexibility when configuring remote configs. Allowing a subset of chains to be updated in a single changeset

### CODE REVIEW

File: deployment/ccip/changeset/cs_update_rmn_config.go [hunk 1]
```diff
@@ -353,14 +357,21 @@ func (c SetRMNRemoteConfig) Validate() error {
...
+		if err != nil {
+			return err
 		}
-	}
 
-	if len(c.Signers) < 2*int(c.F)+1 {
-		return fmt.Errorf("signers count must greater than or equal to %d", 2*c.F+1)
+		for i := 0; i < len(config.Signers)-1; i++ {
+			if config.Signers[i].NodeIndex >= config.Signers[i+1].NodeIndex {
+				return fmt.Errorf("signers must be in ascending order of nodeIndex")
```
Comment (dimkouv) [inline deployment/ccip/changeset/cs_update_rmn_config.go]: nit: include all the node indexes in the error msg

File: deployment/ccip/changeset/cs_update_rmn_config.go [hunk 2]
```diff
@@ -353,14 +357,21 @@ func (c SetRMNRemoteConfig) Validate() error {
...
+			}
+		}
+
+		if len(config.Signers) < 2*int(config.F)+1 {
+			return fmt.Errorf("signers count must greater than or equal to %d", 2*config.F+1)
```
Comment (dimkouv) [inline deployment/ccip/changeset/cs_update_rmn_config.go]: typo: `must BE greater`

Comment (dimkouv) [reply deployment/ccip/changeset/cs_update_rmn_config.go]: also include how many they are `len(config.Signers)`

File: deployment/ccip/changeset/cs_update_rmn_config.go [hunk 3]
```diff
@@ -397,9 +408,10 @@ func NewSetRMNRemoteConfigChangeset(e deployment.Environment, config SetRMNRemot
 
 	rmnRemotePerChain := buildRMNRemotePerChain(e, state)
 	batches := make([]timelock.BatchChainOperation, 0)
-	for chain, remote := range rmnRemotePerChain {
-		if remote == nil {
-			continue
+	for chain, remoteConfig := range config.RMNRemoteConfigs {
+		remote, ok := rmnRemotePerChain[chain]
+		if !ok {
+			return deployment.ChangesetOutput{}, fmt.Errorf("RMNRemote not found for chain %d", chain)
```
Comment (dimkouv) [inline deployment/ccip/changeset/cs_update_rmn_config.go]: maybe change error msg to `RMNRemote config not found...` for clarity

File: deployment/ccip/changeset/cs_update_rmn_config_test.go [hunk 4]
```diff
@@ -166,10 +166,19 @@ func updateRMNConfig(t *testing.T, tc updateRMNConfigTestCase) {
 		signers = append(signers, nop.ToRMNRemoteSigner())
 	}
 
+	remoteConfigs := make(map[uint64]RMNRemoteConfig)
```
Comment (dimkouv) [inline deployment/ccip/changeset/cs_update_rmn_config_test.go]: minor: you can predefine size of the map which is `len(e.Env.Chains)

File: deployment/ccip/changeset/cs_update_rmn_config.go (hunk 3 above)
Comment (carte7000) [reply deployment/ccip/changeset/cs_update_rmn_config.go]: This is actually the contract bindings here. I will precise it in the error message

**Review by rstout — APPROVED** [review]

**Review by dimkouv — APPROVED** [review]

**Pull request merged**